pjrt/convert/trace_container.zig

```diff
@@ -0,0 +1,239 @@
+        if (!device_entry.found_existing) device_entry.value_ptr.* = .{};
+
+        try buildDeviceAndResources(allocator, device_id, xplane, device_entry.value_ptr);
+
+        // Convert events.
+        for (xplane.plane.lines.items) |*xline| {
+            const resource_id: u32 = @intCast(xLineDisplayId(xline));
+
+            if (std.mem.eql(u8, xLineDisplayName(xline), xla_async_op_line_name)) continue;
+            for (xline.events.items) |*xevent| {
+                const event_type = xplane.getEventType(xevent.metadata_id);
+                if (event_type.isInternalEvent()) continue;
+                var event = try self.createEvent(allocator);
+                event.device_id = device_id;
+                event.resource_id = resource_id;
+
+                const metadata = xplane.getEventMetadata(xevent.metadata_id);
+                if (metadata.display_name != .Empty) {
+                    event.name = metadata.display_name.getSlice();
+                    try event.args.put(allocator, "long_name", metadata.name.getSlice());
+                } else {
+                    event.name = metadata.name.getSlice();
+                }
+
+                event.timestamp_ps = (@as(u128, @intCast(xline.timestamp_ns)) * 1000) + @as(u128, @intCast(xevent.data.?.offset_ps));
+                event.duration_ps = @intCast(xevent.duration_ps);
+
+                for (metadata.stats.items) |*xstat| {
+                    if (xstat.value == null) continue;
+                    var stat_buffer = std.ArrayList(u8).init(allocator);
+                    try xstatValueToString(xstat, xplane, stat_buffer.writer().any());
+                    const stat_str = try stat_buffer.toOwnedSlice();
+                    const stat_type = xplane.getStatType(xstat.metadata_id);
+                    if (stat_type.isInternalStat()) continue;
+                    if (stat_type == .step_name) event.name = stat_str;
+                    try event.args.put(allocator, xplane.getStatMetadata(xstat.metadata_id).name.getSlice(), stat_str);
+                }
+
+                for (xevent.stats.items) |*xstat| {
+                    if (xstat.value == null) continue;
+                    var stat_buffer = std.ArrayList(u8).init(allocator);
+                    try xstatValueToString(xstat, xplane, stat_buffer.writer().any());
+                    const stat_str = try stat_buffer.toOwnedSlice();
+                    const stat_type = xplane.getStatType(xstat.metadata_id);
+                    if (stat_type.isInternalStat()) continue;
+                    if (stat_type == .step_name) event.name = stat_str;
+                    try event.args.put(allocator, xplane.getStatMetadata(xstat.metadata_id).name.getSlice(), stat_str);
+                }
+            }
+        }
+    }
+
+    fn getTraceViewerMaxEvents() !u64 {
+        const kMaxEvents = 1000000;
+        if (std.posix.getenv("TF_PROFILER_TRACE_VIEWER_MAX_EVENTS")) |max_events| {
```

Comment:
ugh, we don't do this here. Please move this to an argument to fromXSpace, and if needed to a cli arg